packages/augur-core/source/contracts/libraries/token/ERC1155.sol

```diff
@@ -211,10 +211,12 @@ contract ERC1155 is ERC165, IERC1155 {
     {
         require(ids.length == values.length, "ERC1155: IDs and values must have same lengths");
         require(to != address(0), "ERC1155: target address must be non-zero");
-        require(
-            from == msg.sender || isApprovedForAll(from, msg.sender) == true,
-            "ERC1155: need operator approval for 3rd party transfers"
-        );
+
+        // Bypass approvals for the matic sandbox
```

Comment:
Do the assets being transferred on the predicate essentially just exist as some custom tokens the predicate uses as standins for real tokens that can only be redeemed once a counterparty also agrees that the transaction was valid? I am still missing some significant aspect of this I think. When these contracts transfer Cash or Share tokens what are they intending to actually transfer?

Reply:
In particular I'm wondering about 2 thing: 

1. situations like where these transactions would have failed on their main/matic chain counterparts. Say a user had not approved (or had withdrawn approval) their share tokens for transfer on Matic. I imagine this is handled by the proof mechanism by needing to provide proof that the equivalent transaction would have succeeded?

2. When the real assets need to be distributed how is that done?